### DISCUSSION TITLE
Silence logs from a single crate

### DISCUSSION THREAD

**Discussion by jafioti** [tokio-rs/tracing#1751] [Q&A] [answered]
Hi, sorry if this has been asked before, but is it possible to silence logs from a single crate? I am using hyper, and it's spamming the DEBUG channel, which I would like to use for my application.

**Comment by hawkw** [comment, marked as answer]
You can do this using the [`EnvFilter` layer](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.EnvFilter.html) from `tracing-subscriber`, or its lighter-weight cousin, the [`Targets` layer](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/targets/struct.Targets.html). See [here](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/layer/index.html#filtering-with-layers) for an overview of how to use filter layers with `tracing-subscriber`.

Hope that helps!

**Reply by jafioti** [reply]
Yep, that's great! Thanks for the options.
